src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -286,6 +315,18 @@ def __write_kv_to_app_service(cmd, key_values, appservice_account):
         for kv in key_values:
             name = kv.key
             value = kv.value
+            # If its a KeyVault ref, convert the format to AppService KeyVault ref format
+            if kv.content_type and kv.content_type.lower() == KeyVaultConstants.KEYVAULT_CONTENT_TYPE:
+                try:
+                    secret_uri = json.loads(value).get("uri")
+                    if secret_uri:
+                        value = KeyVaultConstants.APPSVC_KEYVAULT_PREFIX + '(SecretUri={0})'.format(secret_uri)
+                    else:
+                        raise ValueError("Not a valid KeyVault reference.")
+                except (AttributeError, TypeError, ValueError) as e:
```

Comment:
nit: no need to add error message as it will be caught later and swallowed.

Reply:
If value doesn't contain a "uri" field, or the uri value is empty string, we want to treat it as a regular key value and print this message in debugger. Since this case will not throw an exceptoin, to catch this error, we need an else block. So I can either raise a ValueError from else block or duplicate the same logger.debug statement as below.

---
In reply to: [364480722](https://github.com/Azure/azure-cli/pull/11773#discussion_r364480722) [](ancestors = 364480722)

Reply:
I see. Didn't notice you also log the exception in line 333. Then I am fine with throwing Value Error here. 

---
In reply to: [364498010](https://github.com/Azure/azure-cli/pull/11773#discussion_r364498010) [](ancestors = 364498010,364480722)

Reply:
On second thoughts, logging the error in else block will be more efficient than throwing exception and then logging same error through except block.

---
In reply to: [364925012](https://github.com/Azure/azure-cli/pull/11773#discussion_r364925012) [](ancestors = 364925012,364498010,364480722)